### PR TITLE
Fix issue #660

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

**Comment by tpoterba** [comment]
Ready for a look

File: src/main/scala/org/broadinstitute/hail/utils/RichRow.scala [hunk 1]
```diff
@@ -8,7 +8,8 @@ import scala.language.implicitConversions
 class RichRow(r: Row) {
 
   def update(i: Int, a: Any): Row = {
-    val arr = r.toSeq.toArray
+    val arr = Array.tabulate(r.size)(r.get)
+//    val arr = r.toSeq.toArray
```
Comment (cseed) [inline src/main/scala/org/broadinstitute/hail/utils/RichRow.scala]: Remove.

File: src/main/scala/org/broadinstitute/hail/Utils.scala [hunk 2]
```diff
@@ -516,6 +516,8 @@ class RichOption[T](val o: Option[T]) extends AnyVal {
   def contains(v: T): Boolean = o.isDefined && o.get == v
 
   override def toString: String = o.toString
+
+  def flatZip[U](other: Option[U]): Option[(T, U)] = o.flatMap { val1 => other.map(val2 => (val1, val2)) }
```
Comment (cseed) [inline src/main/scala/org/broadinstitute/hail/Utils.scala]: Oddly, this is what Option.zip is supposed to be, except the return type in the code is wrong and it returns a 0- or 1-element list!  I'm not sure about the name.  I don't see where it flattens.  What about `liftedZip`?

File: src/test/scala/org/broadinstitute/hail/methods/ImputeSexSuite.scala [hunk 3]
```diff
@@ -64,33 +63,32 @@ class ImputeSexSuite extends SparkSuite {
           val (_, imputedSexQuery) = s.vds.querySA("if (sa.imputesex.isFemale) 2 else 1")
           val (_, fQuery) = s.vds.querySA("sa.imputesex.Fstat")
 
-          val hailResult = sc.parallelize(s.vds.sampleIdsAndAnnotations.map { case (sample, sa) =>
-            (sample, (imputedSexQuery(sa), fQuery(sa)))
-          })
+          val hailResult = s.vds.sampleIdsAndAnnotations.map { case (sample, sa) =>
+            (sample, (imputedSexQuery(sa).map(_.asInstanceOf[Int]), fQuery(sa).map(_.asInstanceOf[Double] + .0005)))
```
Comment (cseed) [inline src/test/scala/org/broadinstitute/hail/methods/ImputeSexSuite.scala]: I can't quite see why increasing the MAF above and adding 0.0005 correspond to what plink is doing.

**Pull request merged**